packages/components/src/card/types.ts

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import type { CSSProperties } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import type { Props as SurfaceProps } from '../surface/types';
+
+export type SizeOptions = 'xSmall' | 'small' | 'medium' | 'large';
+
+export type Props = SurfaceProps & {
+	/**
+	 * Renders without a border.
+	 *
+	 * @default false
+	 */
+	isBorderless?: boolean;
+	/**
+	 * Renders with rounded corners.
+	 *
+	 * @default true
+	 */
+	isRounded?: boolean;
+	/**
+	 * Determines the amount of padding within the component.
+	 *
+	 * @default 'medium'
+	 */
+	size?: SizeOptions;
+	/**
+	 * Renders with elevation styles (box shadow).
+	 *
+	 * @default false
+	 */
+	isElevated?: boolean;
+};
+
+type BaseSubComponentProps = {
+	/**
+	 * The children elements.
+	 */
+	children: React.ReactNode;
+	/**
+	 * Renders with a light gray background color.
+	 *
+	 * @default false
+	 */
+	isShady?: boolean;
+	/**
+	 * Determines the amount of padding within the component.
+	 *
+	 * @default 'medium'
+	 */
+	size?: SizeOptions;
+};
+
+export type BodyProps = BaseSubComponentProps & {
+	/**
+	 * Determines if the component is scrollable.
+	 *
+	 * @default true
+	 */
+	isScrollable?: boolean;
+};
+
+export type HeaderProps = BaseSubComponentProps & {
+	/**
+	 * Renders without a border.
+	 *
+	 * @default false
+	 */
+	isBorderless?: boolean;
+};
+
+export type FooterProps = HeaderProps & {
```

Comment:
On a second thought, it feels odd that the `FooterProps` extend `HeaderProps`. Wouldn't it make more sense if both `HeaderProps` and `FooterProps` just extend the same parent type?

Reply:
I created a `MarginalSubComponentProps` type in https://github.com/WordPress/gutenberg/pull/32561/commits/5d5ee6c3dd5ea9c11856c519bb6c695bc85d56be